cocaine/detail/secservice.py

```diff
@@ -0,0 +1,96 @@
+#
+# Mostly compy-pasted from cocaine-tools
+#
+from tornado import gen
+from tornado.ioloop import IOLoop
+
+from cocaine.services import Service
+from cocaine.exceptions import CocaineError
+
+
+class SecureServiceError(CocaineError):
+    pass
+
+
+class BasicSecurity(object):
+    pass
+
+
+class Promiscuous(BasicSecurity):
+    @gen.coroutine
+    def fetch_token(self):
+        raise gen.Return('')
+
+
+class TVM(BasicSecurity):
+
+    def __init__(self, client_id, client_secret, name='tvm'):
```

Comment:
I'd rather inject service directly, instead of name.

`def __init__(self, service, client_id, client_secret):`